### PR TITLE
[Dialogs] Add test for legacy dynamic type

### PR DESCRIPTION
**Pull request by codeman7**
## Motivation
In #7446 we added a flag for bypassing the legacy dynamic type system. Test were added for the setter  but not for the side effects of this property. We should test the setting this property has the correct behavior.

## Detailed changes
In adding these test I found a couple issues within Dialogs.
1. When setting `mdc_adjustsFontForContentSizeCategory` it updates the labels but the fonts may be `nil` because they haven't been set. In order to fix this we move setting `mdc_adjustsFontForContentSizeCategory` lower in the `setupAlertView` method to after all of the fonts have been set and after `adjustsFontForContentSizeCategoryWhenScaledFontIsUnavailable` has been set.
2. When setting `adjustsFontForContentSizeCategoryWhenScaledFontIsUnavailable` that should make it so that the fonts don't scale if `mdc_adjustFontForContentSizeCategory` is on, but we may have already scaled the fonts. In that instance we need to recall `updateFonts` to set them back to the correct `ivar` values.

*Note* This is stacked on #7466

### CODE REVIEW

File: components/Dialogs/src/MDCAlertController.m [hunk 1]
```diff
@@ -369,17 +369,18 @@ - (void)viewDidLoad {
...
-  if (_mdc_adjustsFontForContentSizeCategory) {
-    self.alertView.mdc_adjustsFontForContentSizeCategory = YES;
-  }
   self.alertView.titleLabel.text = self.title;
   self.alertView.messageLabel.text = self.message;
   self.alertView.titleFont = self.titleFont;
   self.alertView.messageFont = self.messageFont;
   self.alertView.titleColor = self.titleColor;
   self.alertView.messageColor = self.messageColor;
+  self.alertView.buttonFont = self.buttonFont;  // b/117717380: Will be deprecated
```
Comment (jverkoey) [inline components/Dialogs/src/MDCAlertController.m:378]: Please update this to a GitHub issue url instead.

File: components/Dialogs/src/MDCAlertController.m [hunk 2]
```diff
@@ -388,7 +389,7 @@ - (void)setupAlertView {
     // Avoid reset title color to white when setting it to nil. only set it for an actual UIColor.
     self.alertView.buttonColor = self.buttonTitleColor;  // b/117717380: Will be deprecated
   }
-  self.alertView.buttonFont = self.buttonFont;  // b/117717380: Will be deprecated
+  // self.alertView.buttonFont = self.buttonFont;  // b/117717380: Will be deprecated
```
Comment (jverkoey) [inline components/Dialogs/src/MDCAlertController.m:392]: Please delete this code rather than commenting it out.

**Review by jverkoey — CHANGES_REQUESTED** [review]
There's a lot going on in this PR. Can we break it into smaller pieces?

**Comment by codeman7** [comment]
@jverkoey Opening up seperate PRs to address all of the labels individually.

File: components/Dialogs/src/MDCAlertController.m (hunk 1 above)
Comment (romoore) [reply components/Dialogs/src/MDCAlertController.m:378]: And please use a `TODO` so it can be tracked more easily in the code.

**Pull request closed without merging**